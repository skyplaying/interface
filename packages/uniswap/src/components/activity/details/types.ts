import {
  ConfirmedSwapTransactionInfo,
  ExactInputSwapTransactionInfo,
  ExactOutputSwapTransactionInfo,
  LpIncentivesClaimTransactionInfo,
  OffRampSaleInfo,
  OnRampPurchaseInfo,
  ReceiveTokenTransactionInfo,
  SendTokenTransactionInfo,
  TransactionType,
  TransactionTypeInfo,
  UnknownTransactionInfo,
} from 'uniswap/src/features/transactions/types/transactionDetails'

export type SwapTypeTransactionInfo =
  | ExactInputSwapTransactionInfo
  | ExactOutputSwapTransactionInfo
  | ConfirmedSwapTransactionInfo

export function isOnRampPurchaseTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is OnRampPurchaseInfo {
  return typeInfo.type === TransactionType.OnRampPurchase
}

export function isOffRampSaleTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is OffRampSaleInfo {
  return typeInfo.type === TransactionType.OffRampSale
}

export function isReceiveTokenTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is ReceiveTokenTransactionInfo {
  return typeInfo.type === TransactionType.Receive
}

export function isSendTokenTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is SendTokenTransactionInfo {
  return typeInfo.type === TransactionType.Send
}

export function isSwapTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is SwapTypeTransactionInfo {
  return typeInfo.type === TransactionType.Swap
}

export function isUnknownTransactionInfo(typeInfo: TransactionTypeInfo): typeInfo is UnknownTransactionInfo {
  return typeInfo.type === TransactionType.Unknown
}

export function isLpIncentivesClaimTransactionInfo(
  typeInfo: TransactionTypeInfo,
): typeInfo is LpIncentivesClaimTransactionInfo {
  return typeInfo.type === TransactionType.LPIncentivesClaimRewards
}

export type TransactionParticipantRowProps = {
  onClose: () => void
  address: string
  isSend?: boolean
}
