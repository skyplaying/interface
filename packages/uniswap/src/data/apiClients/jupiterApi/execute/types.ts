import { z } from 'zod'

// Types derived from https://dev.jup.ag/docs/api/ultra-api/execute

const StatusEnum = z.enum(['Success', 'Failed'])

const SwapEventSchema = z.object({
  inputMint: z.string(),
  inputAmount: z.string(),
  outputMint: z.string(),
  outputAmount: z.string(),
})

export const jupiterExecuteResponseSchema = z
  .object({
    status: StatusEnum,
    signature: z.string().optional(),
    slot: z.string().optional(),
    error: z.string().optional(),
    code: z.number(),
    totalInputAmount: z.string().optional(),
    totalOutputAmount: z.string().optional(),
    inputAmountResult: z.string().optional(),
    outputAmountResult: z.string().optional(),
    swapEvents: z.array(SwapEventSchema).optional(),
  })
  .passthrough()

export type JupiterExecuteResponse = z.infer<typeof jupiterExecuteResponseSchema>
