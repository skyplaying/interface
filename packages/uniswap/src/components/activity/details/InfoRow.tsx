import { PropsWithChildren } from 'react'
import { Flex } from 'ui/src'
import { ContentRow } from 'uniswap/src/components/transactions/requests/ContentRow'

export function InfoRow({
  label,
  children,
}: PropsWithChildren & {
  label: string
}): JSX.Element {
  return (
    <ContentRow label={label} variant="body3">
      <Flex centered row gap="$spacing4">
        {children}
      </Flex>
    </ContentRow>
  )
}
