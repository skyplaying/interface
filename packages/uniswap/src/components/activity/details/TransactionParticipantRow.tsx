import { useTranslation } from 'react-i18next'
import { useDispatch } from 'react-redux'
import { Flex } from 'ui/src'
import { CopyAlt } from 'ui/src/components/icons/CopyAlt'
import { Person } from 'ui/src/components/icons/Person'
import { InfoRow } from 'uniswap/src/components/activity/details/InfoRow'
import { TransactionParticipantDisplay } from 'uniswap/src/components/activity/details/TransactionParticipantDisplay'
import { TransactionParticipantRowProps } from 'uniswap/src/components/activity/details/types'
import { ContextMenu, MenuOptionItem } from 'uniswap/src/components/menus/ContextMenuV2'
import { ContextMenuTriggerMode } from 'uniswap/src/components/menus/types'
import { useUniswapContext } from 'uniswap/src/contexts/UniswapContext'
import { pushNotification } from 'uniswap/src/features/notifications/slice'
import { AppNotificationType, CopyNotificationType } from 'uniswap/src/features/notifications/types'
import { setClipboard } from 'uniswap/src/utils/clipboard'
import { isMobileApp } from 'utilities/src/platform'
import { useBooleanState } from 'utilities/src/react/useBooleanState'

export function TransactionParticipantRow({
  address,
  isSend = false,
  onClose,
}: TransactionParticipantRowProps): JSX.Element {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const { navigateToExternalProfile } = useUniswapContext()

  const { value: isContextMenuOpen, setTrue: openContextMenu, setFalse: closeContextMenu } = useBooleanState(false)

  const onCopyAddress = async (): Promise<void> => {
    await setClipboard(address)
    dispatch(
      pushNotification({
        type: AppNotificationType.Copied,
        copyType: CopyNotificationType.Address,
      }),
    )
  }

  const onViewProfile = (): void => {
    navigateToExternalProfile({ address })
    onClose()
  }

  const options: MenuOptionItem[] = [
    {
      label: t('common.copy.address'),
      onPress: onCopyAddress,
      Icon: CopyAlt,
    },
    ...(isMobileApp
      ? [
          {
            label: t('common.view.profile'),
            onPress: onViewProfile,
            Icon: Person,
          },
        ]
      : []),
  ]

  return (
    <InfoRow label={isSend ? t('common.text.recipient') : t('common.text.sender')}>
      <Flex>
        <ContextMenu
          isPlacementAbove
          menuItems={options}
          triggerMode={ContextMenuTriggerMode.Primary}
          isOpen={isContextMenuOpen}
          closeMenu={closeContextMenu}
          openMenu={openContextMenu}
        >
          <TransactionParticipantDisplay address={address} />
        </ContextMenu>
      </Flex>
    </InfoRow>
  )
}
