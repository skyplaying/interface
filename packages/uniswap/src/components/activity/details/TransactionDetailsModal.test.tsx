import { TransactionDetailsInfoRows } from 'uniswap/src/components/activity/details/TransactionDetailsInfoRows'
import {
  TransactionDetailsContent,
  TransactionDetailsHeader,
} from 'uniswap/src/components/activity/details/TransactionDetailsModal'
import { CurrencyInfo } from 'uniswap/src/features/dataApi/types'
import { TransactionDetails } from 'uniswap/src/features/transactions/types/transactionDetails'
import {
  ARBITRUM_DAI_CURRENCY_INFO,
  BASE_CURRENCY,
  ETH_CURRENCY_INFO,
  OPTIMISM_CURRENCY,
  POLYGON_CURRENCY,
  SAMPLE_SEED_ADDRESS_1,
  currencyInfo,
} from 'uniswap/src/test/fixtures'
import { render } from 'uniswap/src/test/test-utils'

jest.mock('uniswap/src/components/menus/ContextMenuV2', () => ({
  ...jest.requireActual('uniswap/src/components/menus/ContextMenuV2.web'),
}))

const mockWalletAddress = (): Address => SAMPLE_SEED_ADDRESS_1
jest.mock('uniswap/src/features/wallet/hooks/useWallet', () => ({
  useWallet: jest.fn().mockReturnValue({
    evmAccount: { address: mockWalletAddress },
  }),
}))

const mockTransaction = {
  id: '9920dbad-ff24-47c8-814a-094566fc45ff',
  chainId: 81457,
  routing: 'CLASSIC',
  from: '0xee814caea14f6cccfeae34fea11d9a2ca6aabb11',
  typeInfo: {
    type: 'approve',
    tokenAddress: '0x2e8b8dafe7faa3aa2bbcd27cda50ebcdfbd8710c',
    spender: '0xf097e7bed97db1bccd9b067a564aca3d4e5da1f4',
  },
  status: 'confirmed',
  addedTime: 1719911758204,
  options: { request: {} },
  hash: 'b568a9e9-bbe7-42fc-ab00-5070186c0600',
  receipt: {
    transactionIndex: 29529,
    blockNumber: 17489,
    blockHash: 'dfd3ad45-78e7-4124-90f2-92758b4499ba',
    confirmedTime: 1719946653408,
    confirmations: 57408,
    gasUsed: 27844,
    effectiveGasPrice: 2941,
  },
} as TransactionDetails

// Function to set up mocks
const getCurrencyInfoForChain = (chainId: number): CurrencyInfo => {
  switch (chainId) {
    case 1: // Mainnet
      return ETH_CURRENCY_INFO
    case 42161: // Arbitrum One
      return ARBITRUM_DAI_CURRENCY_INFO
    case 8453: // Base
      return currencyInfo({ nativeCurrency: BASE_CURRENCY })
    case 10: // Optimism
      return currencyInfo({ nativeCurrency: OPTIMISM_CURRENCY })
    case 137: // Polygon
      return currencyInfo({ nativeCurrency: POLYGON_CURRENCY })
    default:
      return ETH_CURRENCY_INFO // fallback to ETH
  }
}

jest.mock('uniswap/src/features/tokens/useCurrencyInfo', () => ({
  useCurrencyInfo: (currencyIdString: string | undefined): Maybe<CurrencyInfo> => {
    if (!currencyIdString) {
      return null
    }
    const [, chainIdStr] = currencyIdString.split(':')
    if (!chainIdStr) {
      return null
    }
    const chainId = parseInt(chainIdStr, 10)
    return getCurrencyInfoForChain(chainId)
  },
}))

jest.mock('uniswap/src/features/gating/hooks', () => ({
  useDynamicConfigValue: jest
    .fn()
    .mockImplementation(({ defaultValue }: { config: unknown; key: unknown; defaultValue: unknown }) => {
      return defaultValue
    }),
  useFeatureFlag: jest.fn().mockReturnValue(true),
  getFeatureFlag: jest.fn().mockReturnValue(true),
  useExperimentValue: jest.fn().mockReturnValue('CLASSIC'),
}))

jest.mock('uniswap/src/features/language/localizedDayjs', () => ({
  useFormattedDateTime: jest.fn(() => 'January 1, 2023 12:00 AM'),
  FORMAT_DATE_TIME_MEDIUM: 'MMMM D, YYYY h:mm A',
}))

jest.mock('ui/src/loading/Skeleton', () => ({
  Skeleton: (): JSX.Element => <></>,
}))

describe('TransactionDetails Components', () => {
  it('renders TransactionDetailsHeader without error', () => {
    const transactionActions = [
      {
        label: 'Cancel',
        onPress: jest.fn(),
      },
    ]

    const tree = render(
      <TransactionDetailsHeader transactionActions={transactionActions} transactionDetails={mockTransaction} />,
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders TransactionDetailsContent without error', () => {
    const onClose = jest.fn()

    const tree = render(<TransactionDetailsContent transactionDetails={mockTransaction} onClose={onClose} />)

    expect(tree).toMatchSnapshot()
  })

  it('renders TransactionDetailsInfoRows without error with isShowingMore false', () => {
    const onClose = jest.fn()

    const tree = render(
      <TransactionDetailsInfoRows isShowingMore={false} transactionDetails={mockTransaction} onClose={onClose} />,
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders TransactionDetailsInfoRows without error with isShowingMore true', () => {
    const onClose = jest.fn()

    const tree = render(
      <TransactionDetailsInfoRows isShowingMore={true} transactionDetails={mockTransaction} onClose={onClose} />,
    )

    expect(tree).toMatchSnapshot()
  })
})
