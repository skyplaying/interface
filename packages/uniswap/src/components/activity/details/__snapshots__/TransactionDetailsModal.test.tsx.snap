// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetails Components renders TransactionDetailsContent without error 1`] = `
<View
  style={
    {
      "flexDirection": "column",
    }
  }
>
  <View
    cancelable={true}
    collapsable={false}
    disabled={false}
    focusVisibleStyle={{}}
    focusable={true}
    forwardedRef={[Function]}
    jestAnimatedStyle={
      {
        "value": {},
      }
    }
    minPressDuration={0}
    onBlur={[Function]}
    onFocus={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    role="button"
    style={
      {
        "backgroundColor": "transparent",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "flexDirection": "column",
        "opacity": 1,
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "gap": 8,
          "justifyContent": "center",
          "paddingBottom": 32,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 32,
        }
      }
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.2}
        onBlur={[Function]}
        onFocus={[Function]}
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Basel Grotesk",
            "fontSize": 25,
            "fontWeight": "400",
            "lineHeight": 30,
          }
        }
        suppressHighlighting={true}
      />
      <View
        onBlur={[Function]}
        onFocus={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 8,
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          onBlur={[Function]}
          onFocus={[Function]}
          style={
            {
              "color": "rgba(255, 255, 255, 0.65)",
              "fontFamily": "Basel Grotesk",
              "fontSize": 17,
              "fontWeight": "400",
              "lineHeight": 22.1,
            }
          }
          suppressHighlighting={true}
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`TransactionDetails Components renders TransactionDetailsHeader without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "gap": 12,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          }
        }
      >
        <View
          style={
            {
              "bottom": -4,
              "flexDirection": "column",
              "position": "absolute",
              "right": -4,
              "zIndex": 10,
            }
          }
        >
          <View
            style={
              {
                "borderColor": "#131313",
                "borderRadius": 5.76,
                "borderStyle": "solid",
                "borderWidth": 1.6,
                "flexDirection": "column",
                "height": 19.2,
                "overflow": "hidden",
                "width": 19.2,
                "zIndex": 10,
              }
            }
            testID="network-logo"
          >
            <Image
              resizeMode="contain"
              source="mock-asset-BLAST_LOGO.png"
              style={
                {
                  "height": 16,
                  "width": 16,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "column",
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 4,
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Basel Grotesk",
              "fontSize": 17,
              "fontWeight": "400",
              "lineHeight": 22.1,
            }
          }
          suppressHighlighting={true}
        >
          Approved
        </Text>
      </View>
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.4}
        style={
          {
            "color": "rgba(255, 255, 255, 0.65)",
            "fontFamily": "Basel Grotesk",
            "fontSize": 13,
            "fontWeight": "400",
            "lineHeight": 16,
          }
        }
        suppressHighlighting={true}
      >
        January 1, 2023 12:00 AM
      </Text>
    </View>
  </View>
  <View
    aria-expanded={false}
    cancelable={true}
    disabled={false}
    focusable={true}
    minPressDuration={0}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <div>
      <View
        cancelable={true}
        disabled={false}
        focusable={true}
        minPressDuration={0}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "borderBottomLeftRadius": 999999,
            "borderBottomRightRadius": 999999,
            "borderTopLeftRadius": 999999,
            "borderTopRightRadius": 999999,
            "flexDirection": "column",
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          strokeWidth={8}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "rgba(255, 255, 255, 0.65)",
                "height": 20,
                "width": 20,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          vbHeight={4}
          vbWidth={18}
        >
          <RNSVGGroup
            color="rgba(255, 255, 255, 0.65)"
            fill={null}
            propList={
              [
                "fill",
                "strokeWidth",
              ]
            }
            strokeWidth={8}
          >
            <RNSVGPath
              d="M9 3C9.55228 3 10 2.55228 10 2C10 1.44772 9.55228 1 9 1C8.44772 1 8 1.44772 8 2C8 2.55228 8.44772 3 9 3Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            />
            <RNSVGPath
              d="M16 3C16.5523 3 17 2.55228 17 2C17 1.44772 16.5523 1 16 1C15.4477 1 15 1.44772 15 2C15 2.55228 15.4477 3 16 3Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            />
            <RNSVGPath
              d="M2 3C2.55228 3 3 2.55228 3 2C3 1.44772 2.55228 1 2 1C1.44772 1 1 1.44772 1 2C1 2.55228 1.44772 3 2 3Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </div>
  </View>
</View>
`;

exports[`TransactionDetails Components renders TransactionDetailsInfoRows without error with isShowingMore false 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "gap": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "rgba(255, 255, 255, 0.65)",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Network cost
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 4,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#131313",
            "borderRadius": 4.8,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flexDirection": "column",
            "height": 16,
            "overflow": "hidden",
            "width": 16,
            "zIndex": 10,
          }
        }
        testID="network-logo"
      >
        <Image
          resizeMode="contain"
          source="mock-asset-BLAST_LOGO.png"
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "rgba(255, 255, 255, 0.65)",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Transaction
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 4,
          "justifyContent": "center",
        }
      }
    >
      <View
        cancelable={true}
        collapsable={false}
        disabled={false}
        focusVisibleStyle={{}}
        focusable={true}
        forwardedRef={[Function]}
        jestAnimatedStyle={
          {
            "value": {},
          }
        }
        minPressDuration={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        role="button"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "flexDirection": "row",
            "gap": 6,
            "justifyContent": "center",
            "opacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          onBlur={[Function]}
          onFocus={[Function]}
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Basel Grotesk",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 19.5,
            }
          }
          suppressHighlighting={true}
        />
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={16}
          bbWidth={16}
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          strokeWidth={8}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "rgba(255, 255, 255, 0.38)",
                "height": 16,
                "width": 16,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            color="rgba(255, 255, 255, 0.38)"
            fill={null}
            propList={
              [
                "fill",
                "strokeWidth",
              ]
            }
            strokeWidth={8}
          >
            <RNSVGPath
              d="M21 4V9C21 9.552 20.553 10 20 10C19.447 10 19 9.552 19 9V6.41406L11.707 13.707C11.512 13.902 11.256 14 11 14C10.744 14 10.488 13.902 10.293 13.707C9.90197 13.316 9.90197 12.684 10.293 12.293L17.5859 5H15C14.447 5 14 4.552 14 4C14 3.448 14.447 3 15 3H20C20.13 3 20.2601 3.0269 20.3821 3.0769C20.6271 3.1779 20.8221 3.37292 20.9231 3.61792C20.9741 3.73992 21 3.87 21 4ZM20 12C19.447 12 19 12.448 19 13V17C19 18.439 18.439 19 17 19H7C5.561 19 5 18.439 5 17V7C5 5.561 5.561 5 7 5H11C11.553 5 12 4.552 12 4C12 3.448 11.553 3 11 3H7C4.458 3 3 4.458 3 7V17C3 19.542 4.458 21 7 21H17C19.542 21 21 19.542 21 17V13C21 12.448 20.553 12 20 12Z"
              fill={
                {
                  "type": 2,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "rgba(255, 255, 255, 0.65)",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Submitted on
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 4,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.4}
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Basel Grotesk",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 19.5,
          }
        }
        suppressHighlighting={true}
      >
        January 1, 2023 12:00 AM
      </Text>
    </View>
  </View>
</View>
`;

exports[`TransactionDetails Components renders TransactionDetailsInfoRows without error with isShowingMore true 1`] = `
<View
  style={
    {
      "flexDirection": "column",
      "gap": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "rgba(255, 255, 255, 0.65)",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Network cost
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 4,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#131313",
            "borderRadius": 4.8,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flexDirection": "column",
            "height": 16,
            "overflow": "hidden",
            "width": 16,
            "zIndex": 10,
          }
        }
        testID="network-logo"
      >
        <Image
          resizeMode="contain"
          source="mock-asset-BLAST_LOGO.png"
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "rgba(255, 255, 255, 0.65)",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Transaction
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 4,
          "justifyContent": "center",
        }
      }
    >
      <View
        cancelable={true}
        collapsable={false}
        disabled={false}
        focusVisibleStyle={{}}
        focusable={true}
        forwardedRef={[Function]}
        jestAnimatedStyle={
          {
            "value": {},
          }
        }
        minPressDuration={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        role="button"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "flexDirection": "row",
            "gap": 6,
            "justifyContent": "center",
            "opacity": 1,
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          allowFontScaling={true}
          maxFontSizeMultiplier={1.4}
          onBlur={[Function]}
          onFocus={[Function]}
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "Basel Grotesk",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 19.5,
            }
          }
          suppressHighlighting={true}
        />
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={16}
          bbWidth={16}
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          strokeWidth={8}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "rgba(255, 255, 255, 0.38)",
                "height": 16,
                "width": 16,
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            color="rgba(255, 255, 255, 0.38)"
            fill={null}
            propList={
              [
                "fill",
                "strokeWidth",
              ]
            }
            strokeWidth={8}
          >
            <RNSVGPath
              d="M21 4V9C21 9.552 20.553 10 20 10C19.447 10 19 9.552 19 9V6.41406L11.707 13.707C11.512 13.902 11.256 14 11 14C10.744 14 10.488 13.902 10.293 13.707C9.90197 13.316 9.90197 12.684 10.293 12.293L17.5859 5H15C14.447 5 14 4.552 14 4C14 3.448 14.447 3 15 3H20C20.13 3 20.2601 3.0269 20.3821 3.0769C20.6271 3.1779 20.8221 3.37292 20.9231 3.61792C20.9741 3.73992 21 3.87 21 4ZM20 12C19.447 12 19 12.448 19 13V17C19 18.439 18.439 19 17 19H7C5.561 19 5 18.439 5 17V7C5 5.561 5.561 5 7 5H11C11.553 5 12 4.552 12 4C12 3.448 11.553 3 11 3H7C4.458 3 3 4.458 3 7V17C3 19.542 4.458 21 7 21H17C19.542 21 21 19.542 21 17V13C21 12.448 20.553 12 20 12Z"
              fill={
                {
                  "type": 2,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "rgba(255, 255, 255, 0.65)",
          "fontFamily": "Basel Grotesk",
          "fontSize": 15,
          "fontWeight": "400",
          "lineHeight": 19.5,
        }
      }
      suppressHighlighting={true}
    >
      Submitted on
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 4,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.4}
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Basel Grotesk",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 19.5,
          }
        }
        suppressHighlighting={true}
      >
        January 1, 2023 12:00 AM
      </Text>
    </View>
  </View>
</View>
`;
